cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -74,38 +76,32 @@ func runAbapEnvironmentCheckoutBranch(options *abapEnvironmentCheckoutBranchOpti
 	client.SetOptions(clientOptions)
 	pollIntervall := com.GetPollIntervall()
 
-	log.Entry().Infof("Starting to switch branch to branch '%v' on repository '%v'", options.BranchName, options.RepositoryName)
-	log.Entry().Info("--------------------------------")
-	log.Entry().Info("Start checkout branch: " + options.BranchName)
-	log.Entry().Info("--------------------------------")
+	err = checkCheckoutBranchRepositoryConfiguration(*options)
 
-	// Triggering the Checkout of the repository into the ABAP Environment system
-	uriConnectionDetails, errorTriggerCheckout := triggerCheckout(options.RepositoryName, options.BranchName, connectionDetails, client)
-	if errorTriggerCheckout != nil {
-		log.Entry().WithError(errorTriggerCheckout).Fatal("Checkout of " + options.BranchName + " for software component " + options.RepositoryName + " failed on the ABAP System")
+	if options.Repositories != "" && err == nil {
+		err = checkoutBranchesFromConfigFile(options.Repositories, connectionDetails, client, pollIntervall)
 	}
-
-	// Polling the status of the repository import on the ABAP Environment system
-	status, errorPollEntity := abaputils.PollEntity(options.RepositoryName, uriConnectionDetails, client, pollIntervall)
-	if errorPollEntity != nil {
-		log.Entry().WithError(errorPollEntity).Fatal("Status of checkout action on repository" + options.RepositoryName + " failed on the ABAP System")
+	if err == nil && (options.BranchName != "" && options.RepositoryName != "") {
+		err = handleCheckout(abaputils.Repository{Name: options.RepositoryName, Branch: options.BranchName}, connectionDetails, client, pollIntervall)
 	}
-	if status == "E" {
-		log.Entry().Fatal("Checkout of branch " + options.BranchName + " failed on the ABAP System")
+	if err != nil {
+		return fmt.Errorf("Checking out branches failed : %w", err)
 	}
 
-	log.Entry().Info("--------------------------------")
-	log.Entry().Infof("Checkout of branch %v on repository %v was successful", options.BranchName, options.RepositoryName)
-	log.Entry().Info("--------------------------------")
+	log.Entry().Info("-------------------------")
+	log.Entry().Info("All branches were checked out successfully")
 	return nil
 }
 
 func triggerCheckout(repositoryName string, branchName string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (abaputils.ConnectionDetailsHTTP, error) {
-
 	uriConnectionDetails := checkoutConnectionDetails
 	uriConnectionDetails.URL = ""
 	checkoutConnectionDetails.XCsrfToken = "fetch"
 
+	if repositoryName == "" || branchName == "" {
+		return uriConnectionDetails, fmt.Errorf("Failed to trigger checkout: %w", errors.New("Repository and Branch Configuration is empty. Please make sure that you have specified the correct values"))
```

Comment:
```suggestion
		return uriConnectionDetails, fmt.Errorf("Failed to trigger checkout: %w", errors.New("Repository or Branch Configuration is empty. Please make sure that you have specified the correct values"))
```

Reply:
changed it to and/or to include both cases